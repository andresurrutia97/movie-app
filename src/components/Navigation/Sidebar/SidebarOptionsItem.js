import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const SidebarOptionsItem = ({ path, name }) => {
  return (
    <NavLink
      activeClassName="active"
      className="flex items-center"
      to={path}
      exact
    >
      <div className="flag" />
      <span className="ml-5">{name}</span>
    </NavLink>
  );
};

SidebarOptionsItem.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default SidebarOptionsItem;
