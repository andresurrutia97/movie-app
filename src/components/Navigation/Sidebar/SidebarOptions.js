import React from 'react';
import PropTypes from 'prop-types';
import { useAuth0 } from '@auth0/auth0-react';

import SidebarOptionsItem from './SidebarOptionsItem';

const SidebarOptions = ({ routes, close }) => {
  const { logout } = useAuth0();
  return (
    <ul className="w-full h-full mt-10 flex flex-col justify-between text-gray-300">
      <div>
        {routes.map(({ path, name }) => {
          return (
            <li
              className="item-sidebar"
              key={path}
              onClick={close}
              role="presentation"
            >
              <SidebarOptionsItem path={path} name={name} />
            </li>
          );
        })}
      </div>
      <li
        onClick={() => logout()}
        role="presentation"
        className="flex justify-between ml-2 itms-center text-gray-400 py-2 px-5 cursor-pointer"
      >
        Logout
        <span className="material-icons">exit_to_app</span>
      </li>
    </ul>
  );
};

SidebarOptions.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
  close: PropTypes.func.isRequired,
};

export default SidebarOptions;
