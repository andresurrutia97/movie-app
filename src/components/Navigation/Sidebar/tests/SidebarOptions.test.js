import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { useAuth0 } from '@auth0/auth0-react';

import SidebarOptions from '../SidebarOptions';
import SidebarOptionsItem from '../SidebarOptionsItem';

jest.mock('@auth0/auth0-react');

describe('<SidebarOptions />', () => {
  const routes = [
    {
      path: '/',
      name: 'Home',
      component: 'Home',
    },
  ];
  const close = jest.fn();
  const component = (
    <BrowserRouter>
      <SidebarOptions routes={routes} close={close} />
    </BrowserRouter>
  );

  const logoutMock = jest.fn();

  let wrapper;
  beforeEach(() => {
    useAuth0.mockReturnValue({
      logout: logoutMock,
    });
    wrapper = mount(component);
  });

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain a instance of <SidebarOptionsItem/>', () => {
    expect(wrapper.find(SidebarOptionsItem)).toHaveLength(routes.length);
  });

  it('should call the close function when click an item', () => {
    wrapper.children().find('li').at(0).props().onClick();
    expect(close).toBeCalledTimes(1);
  });

  it('should call the logout function when click on logout button', () => {
    wrapper.children().find('li').at(1).props().onClick();
    expect(logoutMock).toBeCalledTimes(1);
  });
});
