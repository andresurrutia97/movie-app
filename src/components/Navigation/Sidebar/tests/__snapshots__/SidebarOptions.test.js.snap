// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarOptions /> should render correctly 1`] = `
<ul
  className="w-full h-full mt-10 flex flex-col justify-between text-gray-300"
>
  <div>
    <li
      className="item-sidebar"
      onClick={[MockFunction]}
      role="presentation"
    >
      <a
        aria-current="page"
        className="flex items-center active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <div
          className="flag"
        />
        <span
          className="ml-5"
        >
          Home
        </span>
      </a>
    </li>
  </div>
  <li
    className="flex justify-between ml-2 itms-center text-gray-400 py-2 px-5 cursor-pointer"
    onClick={[Function]}
    role="presentation"
  >
    Logout
    <span
      className="material-icons"
    >
      exit_to_app
    </span>
  </li>
</ul>
`;
